helix-core/src/main/java/org/apache/helix/common/caches/TaskDataCache.java

```diff
@@ -267,6 +278,11 @@ public void updateWorkflowContext(String resourceName, WorkflowContext workflowC
    * Update context of the Workflow or Job
    */
   private void updateContext(String resourceName, ZNRecord record) {
```

Comment:
I think this may be an overkill by checking every update. We can let it keep updating but just one check for all the context before doing writes to ZK. Otherwise, say I am trying to update one context 10 times, I need to convert to String every time. 

Also we can leverage Java8 stream to do the checks before write to ZK, it could be efficient instead of every update check.

Reply:
Thanks for the suggestion. I will see if we can do the check at the update part

Reply:
Addressed.